src/demo/apps/DialerApp.tsx

```diff
@@ -1,114 +1,91 @@
             } else if (context.id.phone) {
-                this.setState({pendingCall: context as ContactPayload});
+                setPendingCall(context as ContactPayload);
             }
         });
-        const callListener = new fdc3.IntentListener(fdc3.Intents.START_CALL, (context: Payload): void => {
-            if (!this.state.inCall) {
-                this.handleIntent(context as ContactPayload, true);
+        const call = new fdc3.IntentListener(fdc3.Intents.DIAL_CALL, (context: Payload) => {
+            if (!inCall) {
+                handleIntent(context as ContactPayload, true);
             } else if (context.id.phone) {
-                this.setState({pendingCall: context as ContactPayload});
+                setPendingCall(context as ContactPayload);
             }
         });
-        const contextListener = new fdc3.ContextListener((context: Payload): void => {
+        const context = new fdc3.ContextListener((context: Payload) => {
             if (context.type === "contact") {
-                if (!this.state.inCall) {
-                    this.handleIntent(context as ContactPayload, false);
+                if (!inCall) {
+                    handleIntent(context as ContactPayload, false);
                 }
             }
         });
-    }
-
-    public render(): JSX.Element {
-        const pendingCall: ContactPayload = this.state.pendingCall!;
-
-        return (
-            <div>
-                <Number inCall={this.state.inCall} number={this.state.phoneNumber} handleChange={this.onNumberEntry} />
-                { this.state.inCall
-                    ? <CallTimer />
-                    : <Dialer handleKeyPress={this.onDialerEntry} />
-                }
-                <CallButton canCall={this.state.phoneNumber.length > 0} inCall={this.state.inCall} handleClick={this.toggleCall} />
-                <Dialog show={!!pendingCall} title="Replace call?" body={"Hang up and call " + (pendingCall && pendingCall.id.phone) + "?"} options={["No", "Yes"]} handleOption={this.handleDialog} />
-            </div>
-        );
-    }
-
-    private onNumberEntry(phoneNumber: string): void {
-        this.setState({phoneNumber});
-    }
-
-    private onDialerEntry(key: string): void {
-        this.setState({phoneNumber: this.state.phoneNumber + key});
-    }
-
-    private toggleCall(): void {
-        this.setState({inCall: !this.state.inCall});
-    }
-
-    private handleDialog(option: string): void {
-        if (option === "Yes") {
-            this.setState({
-                phoneNumber: this.state.pendingCall!.id.phone!,
-                pendingCall: null
-            });
-        } else {
-            this.setState({pendingCall: null});
-        }
-    }
-
-    private handleIntent(context: ContactPayload, startCall: boolean): void {
-        const phoneNumber: string = context.id.phone!;
+        // Cleanup
+        return () => {
+            dial.unsubscribe();
```

Comment:
My lack of React knowledge is coming out, but when would this cleanup happen? Is it sensible to create/destroy the listeners every time? 

They are stored in memory, so would get cleared on a page navigate/reload anyway, and I can't think of a reason we would still be on the page but not want the listeners.

Reply:
The `React.useEffect` hook is called when the component is mounted (`ComponentDidMount`), the cleanup happens when the component is unmounted (`ComponentWillUnmount`). In this case it doesn't matter as the window closes the listeners are destroyed. But if the component was alongside another component and removed/hidden and the window remained open, it would remain subscribed even when the component isn't visible.